src/startupscriptgenerator/src/node/scriptgenerator.go

```diff
@@ -44,6 +44,50 @@ const NodeWrapperPath = "/opt/node-wrapper/"
 const LocalIp = "0.0.0.0"
 const inspectParamVariableName = "ORYX_NODE_INSPECT_PARAM"
 
+// Checks if the application insights needs to be enabled for the current runtime
+func shouldApplicationInsightsBeConfigured() bool {
+	nodeAppInsightsKeyEnv := os.Getenv("APPINSIGHTS_INSTRUMENTATIONKEY")
+	nodeAppInsightsEnabledEnv := os.Getenv("APPLICATIONINSIGHTS_CODELESS_ENABLED")
```

Comment:
A user can enable/disable application insights via the Portal. Many who instrument their code directly with app insights are already using the ikey env var. If attach were to be enabled by default (as it is Windows), there must be a way to disable attach without breaking an app which is using it already in-code.

Reply:
we can inject AI if:
1) `APPINSIGHTS_INSTRUMENTATIONKEY` is defined
2) We can't find the app insights sdk as a dependency

Reply:
As per my understanding, new requirement is ..
we will inject AI if
APPINSIGHTS_INSTRUMENTATIONKEY and APPLICATIONINSIGHTS_CODELESS_ENABLED are defined  as env variables

If user has a dependency of application insight already in package.json that's great, if not  globally installed application insight sdk will be referenced

Reply:
From the Portal's perspective, it has no idea whether or not the app insights sdk is present, so it needs a way to alert the runtime instance the the user opted out of app insights. This can either be done with this new enablement flag or with the ikey flag. If we opt to just delete the ikey flag, it will impact users with stuff already deployed the using the ikey flag (they would just need to add it back I suppose). I think it is just more clear to separate the definitions of the two, but happy to discuss this further if you disagree.

Reply:
I will also add that windows app services instances use `ApplicationInsightsAgent_EXTENSION_VERSION` to accomplish the same thing.

Reply:
OK, I'm fine with it then if it makes things easier for the portal. We just have a few more cases to consider, e.g. what if `APPLICATIONINSIGHTS_CODELESS_ENABLED` is true but `APPINSIGHTS_INSTRUMENTATIONKEY` doesn't exist?

Reply:
@markwolff please confirm .. as per  my understanding oryx will configure appinsight if and only if both the environment variables are defined, if only one of it is defined we are simply ignoring it.

Reply:
You would not want to do the injection. This should only occur if someone directly changed these environment variables. I believe in the UX we would render app insights as not enabled if this is the case.